src/yaml/yamlUtils.ts

```diff
@@ -308,3 +311,135 @@ export function convertJsonSnippetToYaml(snippetText: string) {
         })
         .join('\n')
 }
+
+function getNumberOfLeftSpaces(text: string) {
+    let numOfLeftSpaces = 0
```

Comment:
non-blocking:

There are multiple ways of doing this, just to share some from SO

```js
'    foo'.search(/\S/);  // 4, index of first non whitespace char

OR

myString.length - myString.trimLeft().length 
```